src/libs/recentcollect.c

```diff
@@ -86,12 +92,58 @@ void init_key_accels(dt_lib_module_t *self)
   dt_accel_register_lib(self, NC_("accel", "jump back to previous collection"), GDK_KEY_k, GDK_CONTROL_MASK);
 }
 
+
 void connect_key_accels(dt_lib_module_t *self)
 {
   GClosure *closure = g_cclosure_new(G_CALLBACK(_goto_previous), (gpointer)self, NULL);
   dt_accel_connect_lib(self, "jump back to previous collection", closure);
 }
 
+void _menuitem_preferences(GtkMenuItem *menuitem, dt_lib_module_t *self)
+{
+  int previous_recentcollect_max_items = recentcollect_max_items;
+
+  GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("recent collections settings"), GTK_WINDOW(win),
+                                                  GTK_DIALOG_DESTROY_WITH_PARENT,
+                                                 _("cancel"), GTK_RESPONSE_NONE,
+                                                 _("save"), GTK_RESPONSE_YES, NULL);
+  dt_prefs_init_dialog_recentcollect(dialog);
+
+#ifdef GDK_WINDOWING_QUARTZ
+  dt_osx_disallow_fullscreen(dialog);
```

Comment:
This causes Mac CI to yell - in order to use `dt_osx_disallow_fullscreen` you have to add atop the file:

```c
#ifdef GDK_WINDOWING_QUARTZ
#include "osx/osx.h"
#endif
```

Reply:
Thanks, added